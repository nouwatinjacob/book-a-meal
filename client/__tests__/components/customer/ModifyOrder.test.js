import React from 'react';
import {
  shallow
} from 'enzyme';
import {
  ModifyOrder,
  mapStateToProps
} from '../../../src/components/customer/ModifyOrder.jsx';
import {
  getAnOrderResponse
} from '../../__mockData__/mockOrder';

const props = {
  getAnOrderAction: jest.fn(),
  modifyOrderAction: () => Promise.resolve(),
  orderDetails: getAnOrderResponse.order,
  orderState: {
    success: true,
    error: {
      response: {
        data: {
          message: 'Error messages'
        }
      }
    }
  }
};

const ids = [1, 2, 3];

describe('ModifyOrder component with success set as true', () => {
  localStorage.setItem('token', customerToken);
  sessionStorage.setItem('ids', ids);
  const setup = () => shallow(<ModifyOrder {...props} />);

  it('should render ModifyOrder component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });

  it(`should call 'handleQuantityChange'
  when quantity is changed`, (done) => {
    const wrapper = setup();

    const event = {
      currentTarget: {
        value: '5'
      },
      preventDefault: jest.fn()
    };

    const handleQuantityChangeSpy = jest.spyOn(
      wrapper.instance(),
      'handleQuantityChange'
    );
    wrapper.instance().handleQuantityChange(event);
    expect(handleQuantityChangeSpy).toHaveBeenCalled();

    done();
  });

  it(`should call 'handleSubmit'
  when modify order button is clicked`, (done) => {
    const wrapper = setup();

    const event = {
      preventDefault: jest.fn()
    };

    const handleSubmitSpy = jest.spyOn(
      wrapper.instance(),
      'handleSubmit'
    );
    wrapper.instance().handleSubmit(event);
    expect(handleSubmitSpy).toHaveBeenCalled();

    done();
  });
});

describe('ModifyOrder component with success set as false', () => {
  const propss = {
    getAnOrderAction: jest.fn(),
    modifyOrderAction: () => Promise.resolve(),
    orderDetails: getAnOrderResponse.order,
    orderState: {
      success: false,
      error: {
        response: {
          data: {
            message: 'Error messages'
          }
        }
      }
    }
  };
  const setup = () => shallow(<ModifyOrder {...propss} />);

  it('should render ModifyOrder component correctly', (done) => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    done();
  });
});
