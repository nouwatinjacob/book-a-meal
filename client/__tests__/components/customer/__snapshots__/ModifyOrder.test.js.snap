// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModifyOrder component with success set as false should render ModifyOrder component correctly 1`] = `
<div
  className="container"
>
  <CatererHeader />
  <div
    className="wrapper"
  >
    <ToastContainer
      autoClose={5000}
      bodyClassName={null}
      className={null}
      closeButton={
        <CloseButton
          ariaLabel="close"
        />
      }
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnHover={true}
      pauseOnVisibilityChange={true}
      position="top-right"
      progressClassName={null}
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    />
    <div
      className="reduced-container"
    >
      <div
        className="row"
      >
        <div
          className="c-medium-6 c-small-12 c-xsmall-12"
          id="pd-0"
        >
          <h3>
            Edit your Order
          </h3>
        </div>
      </div>
      <br />
      <br />
      <div
        className="row"
      >
        <div
          className="c-medium-6 c-small-12 c-xsmall-12"
          id="pd-0"
        />
        <div
          className="c-medium-6 c-small-12 c-xsmall-12 amount"
          id="pd-0"
        >
          <button
            className="button warning"
            onClick={[Function]}
          >
            Modify Order
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ModifyOrder component with success set as true should render ModifyOrder component correctly 1`] = `
<div
  className="container"
>
  <CatererHeader />
  <div
    className="wrapper"
  >
    <ToastContainer
      autoClose={5000}
      bodyClassName={null}
      className={null}
      closeButton={
        <CloseButton
          ariaLabel="close"
        />
      }
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnHover={true}
      pauseOnVisibilityChange={true}
      position="top-right"
      progressClassName={null}
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    />
    <div
      className="reduced-container"
    >
      <div
        className="row"
      >
        <div
          className="c-medium-6 c-small-12 c-xsmall-12"
          id="pd-0"
        >
          <h3>
            Edit your Order
          </h3>
        </div>
      </div>
      <br />
      <br />
      <div
        className="row"
      >
        <div
          className="c-medium-6 c-small-12 c-xsmall-12"
          id="pd-0"
        />
        <div
          className="c-medium-6 c-small-12 c-xsmall-12 amount"
          id="pd-0"
        >
          <button
            className="button warning"
            onClick={[Function]}
          >
            Modify Order
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
